### PR TITLE
fix: Use Atomic Pointer for updating duration metric

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Follow up to https://github.com/cloudquery/plugin-sdk/pull/1757, as it seems we're accessing table metrics in other Go routines (e.g. `resolveResource`)

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: scheduler/scheduler_dfs.go [hunk 1]
```diff
@@ -102,9 +102,10 @@ func (s *syncClient) resolveTableDfs(ctx context.Context, table *schema.Table, c
 	}
 
 	// we don't need any waitgroups here because we are waiting for the channel to close
-	tableMetrics.Duration = time.Since(startTime)
+	duration := time.Since(startTime)
+	tableMetrics.Duration.Store(&duration)
 	if parent == nil { // Log only for root tables and relations only after resolving is done, otherwise we spam per object instead of per table.
-		logger.Info().Uint64("resources", tableMetrics.Resources).Uint64("errors", tableMetrics.Errors).Dur("duration_ms", tableMetrics.Duration).Msg("table sync finished")
+		logger.Info().Uint64("resources", tableMetrics.Resources).Uint64("errors", tableMetrics.Errors).Dur("duration_ms", *tableMetrics.Duration.Load()).Msg("table sync finished")
```
Comment (candiduslynx) [inline scheduler/scheduler_dfs.go:108]:
```suggestion
		logger.Info().Uint64("resources", tableMetrics.Resources).Uint64("errors", tableMetrics.Errors).Dur("duration_ms", duration).Msg("table sync finished")
```
no need for extra atomic op here

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by candiduslynx — APPROVED** [review]

**Review by disq — APPROVED** [review]

File: scheduler/metrics.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ type TableClientMetrics struct {
 	Resources uint64
 	Errors    uint64
 	Panics    uint64
-	Duration  time.Duration
+	Duration  atomic.Pointer[time.Duration]
```
Comment (candiduslynx) [inline scheduler/metrics.go:19]: change equal to use load

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: scheduler/metrics.go (hunk 2 above)
Comment (candiduslynx) [reply scheduler/metrics.go:19]: https://github.com/cloudquery/plugin-sdk/pull/1765/files#diff-cb3db60fb46d339e0932dde09ff6854c037d4513602558ae13cd8c1790619e5eR26

File: scheduler/metrics.go [hunk 3]
```diff
@@ -16,11 +16,24 @@ type TableClientMetrics struct {
...
+	if a == nil && b != nil {
+		return false
+	}
+	if a != nil && b == nil {
+		return false
+	}
+	if a == nil && b == nil {
+		return true
+	}
+	return *a == *b
```
Comment (candiduslynx) [inline scheduler/metrics.go:32]:
```suggestion
	if a == nil {
		return b == nil
	}
	return b != nil && *a == *b
```

**Review by candiduslynx — COMMENTED** [review]
can be simplified

File: scheduler/metrics.go (hunk 3 above)
Comment (candiduslynx) [reply scheduler/metrics.go]: https://go.dev/play/p/XoTGCoYnw0N

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**